scala/private/rule_impls.bzl

```diff
@@ -343,9 +328,23 @@ def _compile_or_empty(ctx, jars, srcjars, buildijar, transitive_compile_jars,
         full_jars = [ctx.outputs.jar],
         ijars = [ctx.outputs.jar])
   else:
-    java_jar = _compile(ctx, jars, srcjars, buildijar, transitive_compile_jars,
-                        jars2labels,
-                        implicit_junit_deps_needed_for_java_compilation)
+    in_srcjars = _srcjar_filetype.filter(ctx.files.srcs)
+    all_srcjars = depset(in_srcjars, transitive = [srcjars])
+    java_srcs = _java_filetype.filter(ctx.files.srcs)
+    sources = _scala_filetype.filter(ctx.files.srcs) + java_srcs
+    _compile_scala(ctx, ctx.outputs.jar, sources, jars, all_srcjars, buildijar,
+                   transitive_compile_jars, ctx.attr.resource_strip_prefix,
+                   ctx.files.resources, ctx.files.resource_jars, jars2labels,
+                   ctx.attr.scalacopts)
+    # compile the java now
```

Comment:
should we add this `if` inside the `try_to_compile_java_jar` or add another indirection? looks more appropriate that after _compile_scala we'll see just a call to the java counterpart

Reply:
Well, I don't like passing three args (buildijar, ctx.outputs.ijar and ctx.outputs.jar) into compile java. Also, we want a lot more refactors. We want to use the API to create an ijar:

https://docs.bazel.build/versions/master/skylark/lib/java_common.html#run_ijar

so, I'd like to punt on this and continue on cleaning up after I have unblocked the ability to make proto and thrift generation a pure rule, and not a macro.

Reply:
ok, merge away...